src/components/TodoContainer.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Link
+} from "react-router-dom";
+import TodoList from './TodoList';
+import Header from './Header';
+import InputTodo from './InputTodo';
+import About from '../pages/About';
+import NotMatch from '../pages/NotMatch';
+import Navbar from './Navbar';
+
+const TodoContainer = () => {
+  const getInitialTodos = () => {
+    const temp = localStorage.getItem('todos');
+    const savedTodos = JSON.parse(temp);
+    return savedTodos || [];
+  };
+
+  const [todos, setTodos] = useState(getInitialTodos());
+
+  const delTodo = (id) => {
+    setTodos([...todos.filter((todo) => todo.id !== id)]);
+  };
+
+  const setUpdate = (updatedTitle, id) => {
+    if (updatedTitle === '') {
+      delTodo(id);
+      return;
+    }
+    const todo = todos.find((todo) => todo.id === id);
+    todo.title = updatedTitle;
+    setTodos([...todos.map((element) => {
+      if (element.id === id) {
+        return todo;
+      }
+      return element;
+    })]);
+  };
+
+  const handleChange = (id) => {
+    setTodos((previousState) => previousState.map((todo) => {
+      if (todo.id === id) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+      return todo;
+    }));
+  };
+
+  const addTodoItem = (title) => {
+    const newTodo = {
+      id: uuidv4(),
+      title,
+      completed: false,
+    };
+    setTodos([...todos, newTodo]);
+  };
+
+  useEffect(() => {
+    const temp = JSON.stringify(todos);
+    localStorage.setItem('todos', temp);
+  }, [todos]);
+
```

Comment:
Nice using `useEffect` to allow perform side effects in components. side effects like fetching data, directly updating the DOM, and timers.